src/icon/README.md

```diff
@@ -0,0 +1,76 @@
+# @dojo/widgets/icon/Icon widget
+
+Dojo 2's `Icon` widget renders an icon.
+
+
+## Example Usage
+
+*Basic Example*
+```typescript
+import Icon from '@dojo/widgets/icon/index';
```

Comment:
When people use our widgets in their own apps they don't need to use `/index`, this is just for our AMD test loaders sadly. I'm happy to let this go as it though and will update all readmes in a follow up PR